experimental/argset-chaining/dgen.py

```diff
@@ -0,0 +1,138 @@
+import random
+
+def random_subset(fact_list, min_count=0, max_count=1):
```

Comment:
You might interested to know that since 3.10, Python supports a rich [type annotation language](https://docs.python.org/3/library/typing.html) out-of-the-box.